### PR TITLE
chore: identify version numbers using a git tag

### LINKED ISSUE #12349: Wrong version displayed in night builds
# Bug Report

Images downloaded from here: https://ci.infra.status.im/job/status-desktop/job/master/ shows `0.9.90` version instead of the actual one.

image

It seems to be coming from here:
https://github.com/status-im/status-desktop/blob/bdc7155a34904e486dbfa6761ddc8616c96b06c3/VERSION#L1

### PR DESCRIPTION
**Pull request by caybro**
### What does the PR do

- remove VERSION file

Fixes #12349

### Affected areas

Settings/About, version number

### Architecture compliance

- [x] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Local build:
image

### CODE REVIEW

**Comment by caybro** [comment]
@yakimant hi, will need a little help here: it seems like Jenkins is relying some checks on the existence of the VERSION file. The goal of this PR is to get rid of it, and use the git tags only to identify the version number.

**Comment by igor-sirotin** [comment]
@caybro I've pushed a fix for `Jenkinsfile.linux`, let's see if it works and update other jenkinsfiles after.
Also modified the Makefile a bit to use version scripts rather than shell commands.

**Comment by igor-sirotin** [comment]
@status-im/devops looks like git tags are not available on status-desktop CI. I'm not sure how to fix this, can you please take a look?
image

https://ci.status.im/blue/organizations/jenkins/status-desktop%2Fprs%2Flinux%2Fx86_64%2Fpackage/detail/PR-16652/5/pipeline

**Review by jakubgs — APPROVED** [review]
In order for this to work we need to fetch tags when cloning the repo, which is not the default behavior.

We need to add `Advanced clone behaviours` and enable `Fetch tags` for a given job:

image

Which I have done for all jobs I could find.

But I wonder why it's called 3 times:
```
[Pipeline] sh
+ ./scripts/version.sh
[Pipeline] sh
+ ./scripts/version.sh
[Pipeline] sh
+ ./scripts/version.sh
```

**Comment by caybro** [comment]
> @caybro I've pushed a fix for `Jenkinsfile.linux`, let's see if it works and update other jenkinsfiles after. Also modified the Makefile a bit to use version scripts rather than shell commands.

Great, thanks! But it looks like it fails everywhere except the regular Linux build :(

File: Makefile [hunk 1]
```diff
@@ -251,6 +250,16 @@ ifeq ($(OUTPUT_CSV), true)
   $(shell touch .update.timestamp)
 endif
 
+##
+## Versioning
+## 
+
+version:
```
[the remark this answers is not in the document]
Comment (igor-sirotin) [reply Makefile:257]: @jakubgs I added just for local development. Easy to learn&share the version you have right now.

**Comment by igor-sirotin** [comment]
> Great, thanks! But it looks like it fails everywhere except the regular Linux build :(

@caybro well that was my intention, I only fixed Linux build to test:

> I've pushed a fix for Jenkinsfile.linux, let's see if it works and update other jenkinsfiles after.

Since it's working now, I'll fix the others 🙂

**Comment by igor-sirotin** [comment]
> But I wonder why it's called 3 times:
> 
> ```
> [Pipeline] sh
> + ./scripts/version.sh
> [Pipeline] sh
> + ./scripts/version.sh
> [Pipeline] sh
> + ./scripts/version.sh
> ```

@jakubgs I guess because this script is called as non-target to calculate the `DESKTOP_VERSION` variable.
https://github.com/status-im/status-desktop/blob/819ea206fa71657032cf9127f86b8e29a3ceea66/Makefile#L239

**Comment by jakubgs** [comment]
No, those logs represent separate calls via `Jenkinsfile`, not from `Makefile`.

**Review by igor-sirotin — APPROVED** [review]

**Review by jrainville — APPROVED** [review]
Looks good, but let's wait for the release to be published with the old way first just in case it breaks anything. we can do some tests during the next weeks to see if publishing a release with those changes still work

**Review by alexjba — APPROVED** [review]

**Comment by igor-sirotin** [comment]
I've pushed a `v2.33.0-dev` tag. Rebased on top and tested:
```
> make version
v2.33.0-dev-4-gac92d729b
```
(Read as: based on `v2.33.0-dev` with 4 commits on top, last commit ac92d729b.

image

**Comment by siddarthkay** [comment]
seems like a function used in publishing release assets still references the Version file.
https://github.com/status-im/status-desktop/blob/fe759f3599ee5d4ce12dcf732ee7d897b43a1fe4/ci/Jenkinsfile.combined#L89
Is probably a reason why E2E job has failed on Desktop CI.

**Comment by siddarthkay** [comment]
so this bit needs to be updated : 
https://github.com/status-im/status-jenkins-lib/blob/f8444e2370e268b0cfcf1ad0fc5e4a8d7b3c87de/vars/utils.groovy#L10-L23
and a newer version of status-jenkins-lib needs to be published to fix the issue with missing Version number

**Pull request merged**